src/vm/comutilnative.cpp

```diff
@@ -3159,3 +3159,76 @@ INT32 QCALLTYPE CoreFxGlobalization::HashSortKey(PCBYTE pSortKey, INT32 cbSortKe
     return retVal;
 }
 #endif //FEATURE_COREFX_GLOBALIZATION
+
+static MethodTable * g_pStreamMT;
+static WORD g_slotBeginRead, g_slotEndRead;
+static WORD g_slotBeginWrite, g_slotEndWrite;
+
+static bool HasOverriddenStreamMethod(MethodTable * pMT, WORD slot)
+{
+    CONTRACTL{
+        NOTHROW;
+        GC_NOTRIGGER;
+        MODE_ANY;
+        SO_TOLERANT;
+    } CONTRACTL_END;
+
+    PCODE actual = pMT->GetRestoredSlot(slot);
+    PCODE base = g_pStreamMT->GetRestoredSlot(slot);
+    if (actual == base)
+        return false;
+
+    if (!g_pStreamMT->IsZapped())
```

Comment:
`zap` was the codename for ngen when it was built for .NET Framework v1.0 originally. E.g. `src\zap` is where most of the ngen compiler specific code lives.